platform/mds/mds/src/main/java/org/motechproject/mds/repository/MotechDataRepository.java

```diff
@@ -70,6 +70,12 @@ public PersistenceManager getPersistenceManager() {
         return pm;
     }
 
+    public void evictAll() {
+        if (persistenceManagerFactory != null) {
+            persistenceManagerFactory.getDataStoreCache().evictAll();
```

Comment:
I was thinking that when we needed a method to evict one specific entity
we'd add it then.
Do you see a need right now?

On Mon, Oct 12, 2015 at 4:03 AM, Paweł Gesek <notifications@github.com>
wrote:

> In
> platform/mds/mds/src/main/java/org/motechproject/mds/repository/MotechDataRepository.java
> <https://github.com/motech/motech/pull/117#discussion_r41745398>:
>
> > @@ -70,6 +70,12 @@ public PersistenceManager getPersistenceManager() {
> >          return pm;
> >      }
> >
> > +    public void evictAll() {
> > +        if (persistenceManagerFactory != null) {
> > +            persistenceManagerFactory.getDataStoreCache().evictAll();
>
> should we evict everything? Can we add add a second method that evicts the
> records for this entity only, not the entire cache?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech/motech/pull/117/files#r41745398>.
>



-- 
Frank Huster
Mobile Health Innovations
Grameen Foundation
+1.206.325.6690 x201